### PR TITLE
Added error code utility workflow 

### LINKED ISSUE #79: [CI] add error codes utility runner workflow
**Current Behaviour**
Error codes utility runner workflow is not present.

**Desired Behavious**
Add the error codes utility runner workflow.

**Instructions in https://github.com/meshery/meshery-linkerd/issues/213**

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: metonymic-smokey <ahuja.aditi@gmail.com>

**Description**

This PR fixes #79

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by metonymic-smokey** [comment]
@Blakelist7 could you please review this?

Thanks!

File: .github/workflows/error-codes-updater.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  push:
+    branches:
+      - 'master'
+    paths:
+      - '**.go'
+
+jobs:
+  Update-error-codes:
+    name: Error codes utility
+    if: github.repository == 'layer5io/meshsync'
```
Comment (Blakelist7) [inline .github/workflows/error-codes-updater.yml:12]:
```suggestion
    if: github.repository == 'meshery/meshsync'
```

File: .github/workflows/error-codes-updater.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          repository: 'meshery/meshery'
+          # token with write access to meshery repository
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          path: 'meshery'
+          ref: 'master'
+
+      - name: Update and push docs
+        run: |
+          echo '{ "errors_export": "" }' | jq --slurpfile export ./helpers/errorutil_errors_export.json '.errors_export = $export[0]' > ./meshery/docs/_data/errorref/meshkit_errors_export.json
```
Comment (Blakelist7) [inline .github/workflows/error-codes-updater.yml:52]:
```suggestion
          echo '{ "errors_export": "" }' | jq --slurpfile export ./helpers/errorutil_errors_export.json '.errors_export = $export[0]' > ./meshery/docs/_data/errorref/meshsync_errors_export.json
```

File: .github/workflows/error-codes-updater.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          cd ./meshery
+          git config user.name l5io
+          git config user.email ci@layer5.io
+          if git diff --exit-code --quiet
+          then
+            echo "No changes to commit"
+            exit
+          fi
+          git add ./docs/_data/errorref/meshkit_errors_export.json
```
Comment (Blakelist7) [inline .github/workflows/error-codes-updater.yml:61]:
```suggestion
          git add ./docs/_data/errorref/meshsync_errors_export.json
```

File: .github/workflows/error-codes-updater.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          git commit -m "[Docs] Error Code Reference: Updated codes for MeshKit library" --signoff
```
Comment (Blakelist7) [inline .github/workflows/error-codes-updater.yml:62]:
```suggestion
          git commit -m "[Docs] Error Code Reference: Updated codes for MeshSync" --signoff
```

**Review by Blakelist7 — CHANGES_REQUESTED** [review]
@metonymic-smokey looks good to me! Just some changes suggested. : )
But first, let's ask the expert (@DelusionalOptimist) :P

File: .github/workflows/error-codes-updater.yml (hunk 1 above)
Comment (Blakelist7) [reply .github/workflows/error-codes-updater.yml:12]: I guess this is the org/repo name in which the workflow will run.

File: .github/workflows/error-codes-updater.yml (hunk 3 above)
Comment (metonymic-smokey) [reply .github/workflows/error-codes-updater.yml:61]: In these, I let it remain as meshkit errors thinking that's describing the style of errors used(meshkit errors), not the repo.
But I think you're right here

File: .github/workflows/error-codes-updater.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          ref: 'master'
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ secrets.GO_VERSION }}
+
+      - name: Run utility
+        run: |
+          go run github.com/layer5io/meshkit/cmd/errorutil -d . update --skip-dirs meshery -i ./helpers -o ./helpers
```
Comment (DelusionalOptimist) [inline .github/workflows/error-codes-updater.yml:28]:
```suggestion
          go get github.com/layer5io/meshkit/cmd/errorutil
          go run github.com/layer5io/meshkit/cmd/errorutil -d . update --skip-dirs meshery -i ./helpers -o ./helpers
```
@metonymic-smokey let's also add this here to ensure that we use a newer version of the utility on the action runner and not v2.14

**Review by DelusionalOptimist — APPROVED** [review]
Looks good to go. Thanks @metonymic-smokey and @Blakelist7  :rocket:

**Pull request merged**